### ISSUE TITLE
Request to add modules for the RFDK - Render Farm Deployment KIT

### ISSUE THREAD

**Issue by queglay** [hashicorp/terraform-provider-aws#19477]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I was very excited to see the RFDK for cloud rendering, but not having the interconnect with terraform has initially convinced me to build my own.  Still, it would be amazing to see the RFDK integrated as part of the AWS provider plugins.  I would definitely migrate my existing workflow to this if it was possible.

### References

* https://docs.aws.amazon.com/rfdk/latest/guide/what-is-rfdk.html

**Comment by ewbankkit** [comment]
@queglay Thanks for raising this issue.
As the RFDK integrates today with the AWS CDK do you think an integration with the [CDK for Terraform](https://github.com/hashicorp/terraform-cdk) would be more appropriate?

**Comment by queglay** [comment]
Thanks @ewbankkit for your reply.  Is it possible to integrate for both without double the work?  Worded another way, does the current AWS provider for Terraform have some type of translation to satisfy the needs of the CDK and HCL implementations more easily? perhaps @antonbabenko might have some advice here if he is willing to comment...

If I had to chose one, I can appreciate making the decision may be a challenge.  I'm sure some would appreciate the CDK with a VFX background, but I'm not likely to migrate to CDK myself.  I spend half my time writing Python (Artist tools), and the other half in HCL or BASH, and here are a few reasons why I may continue to prefer HCL for describing large infrastructure:

- Most of the problems I had to solve for cloud infra were already largely solved in HCL examples in the community.  Deadline Integration represents about 15% of the problem set and effort, so if RFDK only used the CDK it would not be a big enough chunk of the solution for me to change to the CDK.
- As a declarative format, HCL is great at describing a clear backbone that other functions can attach to in my opinion.  If I tried to write large infra in python, I would find it more work to keep it structured well, and if I created a poor foundation it would be more difficult to correct.  HCL taught me good habits when I was still learning.
- I find HCL makes it possible to extend infra easily with many extra services because of the wide user base and modules available.  Two large problems that consumed most of my time over the last 3 years were in learning how to automate openVPN well and dynamic secrets management (presently with Hashicorp Vault).  HCL reduced the effort to integrate these a great deal, and other community examples helped with many other small problems.
- CDK for terraform seems new, I'm unsure how production hardened it is.
